### PR TITLE
Feature/footer

### PR DESCRIPTION
**Pull request by CourtneyGosselin**
## Ticket(s)
[DIG-1350](https://candig.atlassian.net/browse/DIG-1350)

## Related PR
[CanDIGv2 PR](https://github.com/CanDIG/CanDIGv2/pull/402)

## Description
Addition of footer to all pages that is responsive to sizing. The footer should include:
- CanDIG website - [CanDIG](https://www.distributedgenomics.ca/) 
- CanDIG GitHub - [CanDIG](https://github.com/CanDIG) 
- CanDIG Platform version number - auto populate from whichever version is deployed
- MoHCCN project link & logo - [Home](https://www.marathonofhopecancercentres.ca/)  
- Funding statement: "CanDIG Receives Funds From:"
     - [TFRI](https://www.tfri.ca/)
     - [UHN DATA](https://uhndata.io/)
     - [BCGSC](https://www.bcgsc.ca/)
     - [C3G](https://computationalgenomics.ca/)

### After PR

screencapture-candig-docker-internal-5080-summary-2024-01-30-18_08_54

## Types of Change(s)

-   [ ] 🪲 Bug fix (non-breaking change that fixes an issue)
-   [x] ✨ New feature (non-breaking change that adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Has it been tested for:

-   [x] My change requires a change to the documentation
-   [x] I have updated the documentation accordingly
-   [x] Prettier linter doesn't return errors
-   [ ] Production branch PR browser testing: Chrome, Firefox, Edge, etc.
-   [x] Locally tested
-   [ ] Dev server tested
-   [ ] Production tested when merging into stable/production branch
-   [x] [Runbook tasks](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) pass locally/on UHN-Dev
-   [x] If visuals have changed, [Runbook](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) has been updated with new screenshots

[DIG-1350]: https://candig.atlassian.net/browse/DIG-1350?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/layout/MainLayout/Footer/index.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+                                <IconMail stroke={1.5} size="1.3rem" style={{ color: theme.palette.primary.main, marginRight: '0.5em' }} />
+                                <span style={linkText}>info@distributedgenomics.ca</span>
+                            </Tooltip>
+                        </ButtonBase>
+                    </HelpContainer>
+                </ResponsiveCanDIGContainer>
+            </ResponsiveSocialsContainer>
+            <ResponsiveFundingContainer style={{ marginRight: '1em' }}>
+                <div style={{ marginRight: '1em' }}>
+                    <p style={{ ...linkText, fontWeight: 'bold', marginBottom: '0.5em' }}>CanDIG Receives Funds from:</p>
```
Comment (mshadbolt) [inline src/layout/MainLayout/Footer/index.js]:
```suggestion
                    <p style={{ ...linkText, fontWeight: 'bold', marginBottom: '0.5em' }}>CanDIG receives funds from:</p>
```

**Review by mshadbolt — CHANGES_REQUESTED** [review]
pulled the branch and rebuilt stack. Looked at in Brave and Chrome - both chromium based browsers

A couple of minor bits of feedback:
* When the summary page loads on my screen, the footer is out of sight of the main page as there is a large gap under the graphs. 
Screenshot 2024-01-31 at 3 17 46 PM
* Maybe we could reduce this spacing so that the footer shows at the bottom of the page. So minimum spacing would be about the same as the gap at the top with the header but when there is enough space it is anchored to the bottom of the screen.
Like:
Screenshot 2024-01-31 at 3 17 46 PM-edits

* Can we change the capitalisation of the funding statement to: "CanDIG receives funds from:" (sorry I pointed this out  earlier on the shared screenshots)
* Screenshots above don't show it but I played with the CANDIG_VERSION to see that it displays and it worked for me, not sure if it could be confusing about whether this is the version of the entire stack or the data portal itself.
* Would be nice for the Candig version statement link to the relevant version of the stack? should be easy to add the given version to the url `https://github.com/CanDIG/CanDIGv2/releases/tag/v2.1.1` -> `https://github.com/CanDIG/CanDIGv2/releases/tag/${CANDIG_VERSION}` (however this is done in js that I have no clue about ) this means we do have to ensure the version is input into the .env file correctly when making stable releases to make sure the link isn't broken

File: src/layout/MainLayout/Footer/index.js (hunk 1 above)
Comment (CourtneyGosselin) [reply src/layout/MainLayout/Footer/index.js]: Changes were made and responsiveness of graphs moved to their own ticket https://candig.atlassian.net/jira/software/c/projects/DIG/boards/6/backlog?selectedIssue=DIG-1470

**Review by mshadbolt — APPROVED** [review]
Thanks for making the updates! Happy to approve this from a visuals point of view (know nothing about the code so will leave that up to @OrdiNeu )

File: src/layout/MainLayout/Footer/index.js [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+    ...flexFrame,
+    alignItems: 'center'
+};
+
+const linkText = {
+    margin: 0,
+    fontWeight: 'bold'
+};
+
+const SocialsContainer = styled(Box)(({ theme }) => ({
```
Comment (OrdiNeu) [inline src/layout/MainLayout/Footer/index.js]: ESLint warning
```suggestion
const SocialsContainer = styled(Box)(({ _ }) => ({
```

File: src/layout/MainLayout/Footer/index.js [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+        ...linkFrame,
+        textDecoration: 'none',
+        color: theme.palette.primary.main,
+        '&:visited': {
+            color: theme.palette.primary.main
+        }
+    }
+}));
+
+const LinkFrame = styled(Box)(({ theme }) => ({
```
Comment (OrdiNeu) [inline src/layout/MainLayout/Footer/index.js]: ESLint warning
```suggestion
const LinkFrame = styled(Box)(({ _ }) => ({
```

File: src/layout/MainLayout/Footer/index.js [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+}));
+
+const HelpContainer = styled(Box)(({ theme }) => ({
+    ...flexFrame,
+    flexDirection: 'column',
+    alignItems: 'flex-start',
+    color: theme.palette.primary.main
+}));
+
+const ResponsiveFooterContainer = styled('footer')(({ theme }) => ({
```
Comment (OrdiNeu) [inline src/layout/MainLayout/Footer/index.js:69]: Could be simplified with the classNames thing that the rest of the components use, but probably not neccecssary

File: src/layout/MainLayout/Footer/index.js [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+
+    function fallbackCopyTextToClipboard(text) {
+        const textarea = document.createElement('textarea');
+        textarea.value = text;
+        document.body.appendChild(textarea);
+        textarea.select();
+
+        try {
+            document.execCommand('copy');
+            console.log('Email copied to clipboard');
```
Comment (OrdiNeu) [inline src/layout/MainLayout/Footer/index.js]:
```suggestion
```

File: src/layout/MainLayout/Footer/index.js [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+        }
+    }
+
+    function copyEmail(email) {
+        try {
+            if (navigator.clipboard) {
+                navigator.clipboard
+                    .writeText(email)
+                    .then(() => {
+                        console.log('Email copied to clipboard');
```
Comment (OrdiNeu) [inline src/layout/MainLayout/Footer/index.js]:
```suggestion
```
NB: This is usually a mailto: redirect (I believe usually for accessibility?), but to be honest I prefer this copying

**Review by OrdiNeu — APPROVED** [review]
Looks good to me

File: src/layout/MainLayout/Footer/index.js (hunk 6 above)
Comment (CourtneyGosselin) [reply src/layout/MainLayout/Footer/index.js]: mailto bug not working on all browsers fix for the meantime till that can be fixed for accessibility. This atleast covers dexterity issues when copying but may not be ideal for screen readers. I will add a ticket to JIRA

**Pull request merged**